### DISCUSSION TITLE
Some git repositories need more than 7 digits to distinguish commits.

### DISCUSSION THREAD

**Discussion by bc-lee** [renovatebot/renovate#22000] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

Currently, renovate displays digests changes with 7 digits.
However, some git repositories with long histories need more than 7 digits
to distinguish commits. For example, [chromium](https://chromium.googlesource.com/chromium/src) repository has 1.2M commits,
and 7 digits aren't enough.
(Currently, `$ git rev-parse --short` displays 13 digits.)

It seems like adding a flag to whether display digests in full format or not
is straightforward. (I actually implemented it in my local environment.)

Maintainers, do you think it is okay to send a PR for this?

**Comment by rarkins** [comment]
How is this a problem today? Are you describing a cosmetic change or functional?

**Reply by bc-lee** [reply]
> How is this a problem today? Are you describing a cosmetic change or functional?

It is both cosmetic changes and functional changes. The comment which renovate bot creates doesn't have enough information, so I'd like to parse the PR message and do additional things.
However, given that renovate bot gives only 7 digits of the commit hash, it may not be reliable to identify the commit. I'd like to have the full commit hash.

What I'm doing is here: [renovate-git-history-action](https://github.com/bc-lee/renovate-git-history-action)
Usage is here: [Link](https://github.com/bc-lee/renovate-git-dep-demo/pull/1#issuecomment-1537071786)

**Reply by rarkins** [reply]
If you are taking cosmetic elements (Eg PR title or commit message) and using them functionally, that doesn't make it functional for the Renovate context. If Renovate internally is making incorrect decisions or comparisons then that's functional.

**Reply by rarkins** [reply]
I want to avoid making the default user experience worse/more verbose for 99.99% of users only to satisfy the interests of one edge use case. If we can make changes to allow you to achieve what you want "opt in" then that's much preferred. You may be able to make these already via PR templating

**Reply by bc-lee** [reply]
Okay. Thanks for explaining it. It shouldn't affect almost every other user's experience worse since that option won't be turned on by default. Maybe you can see my PR and decide whether to merge it or not. Thanks.

**Reply by rarkins** [reply]
Which PR? Which aspect?
